src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -357,7 +357,7 @@ public InsertValidationResponse insertRows(
     this.channelState = channelRuntimeState;
     this.channelFullyQualifiedName = fullyQualifiedChannelName;
     this.nonNullableFieldNames = new HashSet<>();
-    this.flushLock = new ReentrantLock();
+    this.flushLock = new ReentrantLock(true);
```

Comment:
Is there any perf implication with this change?

Reply:
There could be perf implication main when we are doing the expected things.
When the customer have threads waiting for the lock and a new thread comes to acquire the lock, for unfairness lock, we would directly give the lock to the new thread; while for fairness lock, we would need to wait for the first come thread to wake up --  meaning that the performance implication happens when we are not inserting the data according to the customer's order.